integration-tests/src/test/java/oracle/kubernetes/operator/ItPodsShutdownMS.java

```diff
@@ -0,0 +1,436 @@
+        TestUtils.deleteWeblogicDomainResources(domain.getDomainUid());
+      }
+    }
+    LoggerHelper.getLocal().log(Level.INFO, "SUCCESS - {0}", testMethodName);
+  }
+
+  /**
+   * Start domain with added shutdown options at the managed server level with IgnoreSessions=false
+   * and verify values are propagated to specified server level. Server shuts down only after sessions are ended.
+   *
+   * @throws Exception if domain cannot be started or failed to verify shutdown options.
+   */
+  @Test
+  public void testAddShutdownOptionsToMsIgnoreSessions() throws Exception {
+
+    Assumptions.assumeTrue(FULLTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethodName);
+
+    Map<String, Object> shutdownProps = new HashMap();
+    shutdownProps.put("timeoutSeconds", 160);
+    shutdownProps.put("ignoreSessions", false);
+    List<Map<String,Object>> shutdownPropsMSs = new ArrayList<>();
+    Map<String, Object> shutdownPropsMyMS = new HashMap();
+    shutdownPropsMyMS.put("managed-server1", shutdownProps);
+    shutdownPropsMSs.add(shutdownPropsMyMS);
+    Map<String, Object> shutdownPropOpt = new HashMap();
+    shutdownPropOpt.put("server",shutdownPropsMSs);
+    Domain domain = null;
+    try {
+      domain = createDomain(domainNSShutOpMSIgnoreSessions, shutdownPropOpt);
+      Assertions.assertNotNull(domain,
+          domainNSShutOpMSIgnoreSessions
+              + " failed to create domain " + domainNSShutOpMSIgnoreSessions);
+      Assertions.assertTrue(
+          checkShutdownProp(domain.getDomainUid()
+              + "-managed-server1", domain.getDomainNs(),"160", "false"),
+          domain.getDomainUid()
+              + "-managed-server1: "
+              + " shutdown property does not match the expected :"
+              + " timeoutSeconds=160, ignoreSessions=false");
+      domain.buildDeployJavaAppInPod(
+          testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword());
+      domain.callWebAppAndVerifyLoadBalancing(testAppName + "/CounterServlet?", false);
+
+      long delayTime = 80 * 1000;
+      long terminationTimeWithIgnoreSessionFalse = checkShutdownTime(delayTime, domain);
+
+      if (terminationTimeWithIgnoreSessionFalse < delayTime) {
+        LoggerHelper.getLocal().log(Level.INFO, "FAILURE: ignored opened session during shutdown");
+        throw new Exception("FAILURE: ignored opened session during shutdown");
+      }
+      LoggerHelper.getLocal().log(Level.INFO,
+          " Termination time with ignoreSession=false :" + terminationTimeWithIgnoreSessionFalse);
+    } finally {
+      if (domain != null) {
+        TestUtils.deleteWeblogicDomainResources(domain.getDomainUid());
+      }
+    }
+    LoggerHelper.getLocal().log(Level.INFO, "SUCCESS - {0}", testMethodName);
+  }
+
+  /**
+   * Add shutdown option Timeout at managed server level and verify all pods are Terminated
+   * according to the setting.
```

Comment:
More Detail:  How the servers are terminated.  The more detail we describe in the Javadoc, it easier to debug the failure by understanding use case. So the Javadoc is the only source of truth.

Reply:
added more details